pkg/watcher/reconciler/dynamic/dynamic.go

```diff
@@ -182,7 +182,7 @@ func (r *Reconciler) deleteUponCompletion(ctx context.Context, o results.Object)
 		return nil
 	}
 
-	if ownerReferences := o.GetOwnerReferences(); len(ownerReferences) > 0 {
+	if ownerReferences := o.GetOwnerReferences(); r.cfg.CheckOwner && len(ownerReferences) > 0 {
```

Comment:
I thought that one of the reason the check was put in place was to avoid deleting taskruns before the pipelineruns that owns them had the time to process the taskrun's logs.
If that's correct, additional logic needs to be added to cancel the deletion in the case one of the ownerRef is of type PipelineRun, even if checkOwner is set to false.

Reply:
This check was implemented in #255 to improve stability in deletion, and it was not related to the processing of taskrun logs. 
For example, consider the situation if only a taskrun is executed (ie. without a parent pipelinerun), in that case there won't be any owner references and hence the controller won't wait for the deletion before log streaming is completed or failed.

Having said that, I agree that there should be additional logic to handle the log streaming situation, but it's not a scope of this PR, as that logic is not related to `ownerRef`. There is a open bug for this, #514.